client/my-sites/checklist/wpcom-checklist/checklist-banner.js

```diff
@@ -0,0 +1,155 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import Gridicon from 'gridicons';
+import PropTypes from 'prop-types';
+import React, { Children, Component } from 'react';
+import store from 'store';
+import { connect } from 'react-redux';
+import { find, get, reduce } from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import Card from 'components/card';
+import ChecklistShowShare from 'my-sites/checklist/share';
+import ChecklistBannerTask from './checklist-banner-task';
+import Gauge from 'components/gauge';
+import ProgressBar from 'components/progress-bar';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getSiteOption, getSiteSlug } from 'state/sites/selectors';
+
+const storeKeyForNeverShow = 'sitesNeverShowChecklistBanner';
+
+export class ChecklistBanner extends Component {
+	static propTypes = {
+		translate: PropTypes.func.isRequired,
+		siteDesignType: PropTypes.string,
+		siteSlug: PropTypes.string,
+	};
+
+	state = { closed: false };
+
+	handleClose = () => {
+		const { siteId } = this.props;
+		const sitesNeverShowBanner = store.get( storeKeyForNeverShow ) || {};
+		sitesNeverShowBanner[ `${ siteId }` ] = true;
+		store.set( storeKeyForNeverShow, sitesNeverShowBanner );
+
+		this.setState( { closed: true } );
+
+		this.props.track( 'calypso_checklist_banner_close', {
+			site_id: siteId,
+		} );
+	};
+
+	canShow() {
+		if ( this.state.closed ) {
+			return false;
+		}
+
+		if ( this.props.siteDesignType !== 'blog' ) {
+			return false;
+		}
+
+		const sitesNeverShowBanner = store.get( storeKeyForNeverShow );
+		if ( get( sitesNeverShowBanner, String( this.props.siteId ) ) === true ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/* eslint-disable wpcalypso/jsx-classname-namespace */
+
+	render() {
+		const { translate } = this.props;
+
+		if ( ! this.canShow() ) {
+			return null;
+		}
+
+		const childrenArray = Children.toArray( this.props.children );
+		const total = childrenArray.length;
```

Comment:
Could use the more direct (semantic?) `Children.count( this.props.children )` here. (Might be a tad stabler API? If they ever take `toArray()` from us). <- Nitpick

Reply:
I opted for this because I already had to `toArray` to calculate completion and find the child to render.